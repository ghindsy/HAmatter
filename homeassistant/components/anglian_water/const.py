"""Anglian Water consts."""

from logging import Logger, getLogger

LOGGER: Logger = getLogger(__package__)

NAME = "Anglian Water"
DOMAIN = "anglian_water"

CONF_DEVICE_ID = "device_id"
