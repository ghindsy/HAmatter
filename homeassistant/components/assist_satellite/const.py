"""Constants for assist satellite."""

DOMAIN = "assist_satellite"
