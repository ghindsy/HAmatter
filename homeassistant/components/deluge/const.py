"""Constants for the Deluge integration."""

import enum
import logging
from typing import Final

CONF_WEB_PORT = "web_port"
CURRENT_STATUS = "current_status"
DEFAULT_NAME = "Deluge"
DEFAULT_RPC_PORT = 58846
DEFAULT_WEB_PORT = 8112
DOMAIN: Final = "deluge"
DOWNLOAD_SPEED = "download_speed"

LOGGER = logging.getLogger(__package__)
PROTOCOL_TRAFFIC_DOWNLOAD_SPEED = "protocol_traffic_download_speed"
PROTOCOL_TRAFFIC_UPLOAD_SPEED = "protocol_traffic_upload_speed"
UPLOAD_SPEED = "upload_speed"


class DelugeGetSessionStatusKeys(enum.Enum):
    """Enum representing the keys that get passed into the Deluge RPC `core.get_session_status` xml rpc method.

    you can call `core.get_session_status` with no keys (so an empty list in deluge-client.DelugeRPCClient.call)
    to get the full list of possible keys, but it seems to basically be a all of the session statistics
    listed on this page: https://www.rasterbar.com/products/libtorrent/manual-ref.html#session-statistics
    and a few others

    there is also a list of deprecated keys that deluge will translate for you and issue a warning in the log:
    https://github.com/deluge-torrent/deluge/blob/7f3f7f69ee78610e95bea07d99f699e9310c4e08/deluge/core/core.py#L58

    """

    DHT_DOWNLOAD_RATE = "dht_download_rate"
    DHT_UPLOAD_RATE = "dht_upload_rate"
    DOWNLOAD_RATE = "download_rate"
    UPLOAD_RATE = "upload_rate"
