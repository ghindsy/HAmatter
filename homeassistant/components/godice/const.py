"""Constants for the godice integration."""

from datetime import timedelta

DOMAIN = "godice"
SCAN_INTERVAL = timedelta(seconds=60)

MANUFACTURER = "Particula"
MODEL = "GoDice"

CONF_SHELL = "shell"

DICE_MIN_NUMBER = 1
DICE_MAX_NUMBER = 20
