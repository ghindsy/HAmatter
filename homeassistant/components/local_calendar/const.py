"""Constants for the Local Calendar integration."""

DOMAIN = "local_calendar"

CONF_CALENDAR_NAME = "calendar_name"
CONF_ICS_FILE = "ics_file"
CONF_IMPORT_ICS = "import_ics"
CONF_STORAGE_KEY = "storage_key"
STORAGE_PATH = ".storage/local_calendar.{key}.ics"
