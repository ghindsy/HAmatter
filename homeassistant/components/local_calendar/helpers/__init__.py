"""Helpers for local_calendar."""
