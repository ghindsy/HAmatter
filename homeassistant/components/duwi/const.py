"""Constants for the Duwi integration.

This module contains constants that are used throughout the Duwi Smart Hub integration,
including configuration keys, default values, and URLs for API endpoints.
"""

from homeassistant.const import Platform

# Unique domain identifier for the Duwi Smart Hub integration
DOMAIN = "duwi"

# Manufacturer of the product for identification purposes
MANUFACTURER = "Duwi"

# Version of the App for the Duwi Smart Hub integration, used for tracking and compatibility.
APP_VERSION = "0.1.1"

# Home Assistant client version
CLIENT_VERSION = "0.1.1"

# Model identification of the Home Assistant client, typically used for logging or diagnostics.
CLIENT_MODEL = "homeassistant"

# List of platforms that support config entry
SUPPORTED_PLATFORMS = [Platform.SWITCH]

SENSOR_TYPE = "7"
