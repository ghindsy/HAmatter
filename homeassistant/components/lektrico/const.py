"""Constants for the Lektrico Charging Station integration."""

from logging import Logger, getLogger

from homeassistant.const import Platform

# Integration domain
DOMAIN = "lektrico"

# Logger
LOGGER: Logger = getLogger(__package__)

# List the platforms that charger supports.
CHARGERS_PLATFORMS = [Platform.SENSOR]

# List the platforms that load balancer device supports.
LB_DEVICES_PLATFORMS = [Platform.SENSOR]
