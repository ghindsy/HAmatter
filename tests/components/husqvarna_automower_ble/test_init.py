"""Test the Husqvarna Automower Bluetooth setup."""

import contextlib
from unittest.mock import Mock

from bleak import BleakError
import pytest
from syrupy.assertion import SnapshotAssertion

from homeassistant.components.husqvarna_automower_ble.const import DOMAIN
from homeassistant.config_entries import ConfigEntryState
from homeassistant.core import HomeAssistant
from homeassistant.helpers import device_registry as dr
from homeassistant.helpers.update_coordinator import UpdateFailed

from . import AUTOMOWER_SERVICE_INFO

from tests.common import MockConfigEntry

pytestmark = pytest.mark.usefixtures("mock_client")


async def test_setup(
    hass: HomeAssistant,
    device_registry: dr.DeviceRegistry,
    mock_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test setup creates expected devices."""

    mock_entry.add_to_hass(hass)
    await hass.config_entries.async_setup(mock_entry.entry_id)
    await hass.async_block_till_done()

    assert mock_entry.state is ConfigEntryState.LOADED

    device = device_registry.async_get_device(
        identifiers={(DOMAIN, AUTOMOWER_SERVICE_INFO.address)}
    )
    assert device == snapshot

    coordinator = mock_entry.runtime_data.coordinator
    await coordinator.async_shutdown()


async def test_setup_retry_connect(
    hass: HomeAssistant,
    device_registry: dr.DeviceRegistry,
    mock_client: Mock,
    mock_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test setup creates expected devices."""

    async def _connect(self, *args, **kwargs) -> bool:
        """Mock BleakClient.connect."""
        return False

    mock_client.connect = _connect

    mock_entry.add_to_hass(hass)
    await hass.config_entries.async_setup(mock_entry.entry_id)
    await hass.async_block_till_done()

    assert mock_entry.state is ConfigEntryState.SETUP_RETRY


async def test_setup_failed_connect(
    hass: HomeAssistant,
    device_registry: dr.DeviceRegistry,
    mock_client: Mock,
    mock_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test setup creates expected devices."""

    async def _connect(self, *args, **kwargs) -> bool:
        """Mock BleakClient.connect."""
        raise BleakError

    mock_client.connect = _connect

    mock_entry.add_to_hass(hass)
    await hass.config_entries.async_setup(mock_entry.entry_id)
    await hass.async_block_till_done()

    assert mock_entry.state is ConfigEntryState.SETUP_RETRY


async def test_setup_disconnect_one(
    hass: HomeAssistant,
    device_registry: dr.DeviceRegistry,
    mock_client: Mock,
    mock_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test setup creates expected devices."""

    async def _connect_success(self, *args, **kwargs) -> bool:
        """Mock BleakClient.connect."""
        return True

    mock_client.connect = _connect_success

    mock_entry.add_to_hass(hass)
    await hass.config_entries.async_setup(mock_entry.entry_id)
    await hass.async_block_till_done()

    assert mock_entry.state is ConfigEntryState.LOADED

    coordinator = mock_entry.runtime_data.coordinator

    def _is_connected(self, *args, **kwargs) -> bool:
        """Mock BleakClient.is_connected."""
        return False

    mock_client.is_connected = _is_connected

    await coordinator._async_update_data()


async def test_setup_disconnect_two(
    hass: HomeAssistant,
    device_registry: dr.DeviceRegistry,
    mock_client: Mock,
    mock_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test setup creates expected devices."""

    async def _connect_success(self, *args, **kwargs) -> bool:
        """Mock BleakClient.connect."""
        return True

    mock_client.connect = _connect_success

    mock_entry.add_to_hass(hass)
    await hass.config_entries.async_setup(mock_entry.entry_id)
    await hass.async_block_till_done()

    assert mock_entry.state is ConfigEntryState.LOADED

    def _is_connected(self, *args, **kwargs) -> bool:
        """Mock BleakClient.is_connected."""
        return False

    async def _connect(self, *args, **kwargs) -> bool:
        """Mock BleakClient.connect."""
        raise BleakError

    mock_client.connect = _connect
    mock_client.is_connected = _is_connected

    coordinator = mock_entry.runtime_data.coordinator

    try:
        await coordinator._async_update_data()
    except UpdateFailed:
        await coordinator.async_shutdown()


async def test_setup_disconnect_three(
    hass: HomeAssistant,
    device_registry: dr.DeviceRegistry,
    mock_client: Mock,
    mock_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test setup creates expected devices."""

    async def _connect_success(self, *args, **kwargs) -> bool:
        """Mock BleakClient.connect."""
        return True

    mock_client.connect = _connect_success

    mock_entry.add_to_hass(hass)
    await hass.config_entries.async_setup(mock_entry.entry_id)
    await hass.async_block_till_done()

    assert mock_entry.state is ConfigEntryState.LOADED

    def _is_connected(self, *args, **kwargs) -> bool:
        """Mock BleakClient.is_connected."""
        return False

    async def _connect(self, *args, **kwargs) -> bool:
        """Mock BleakClient.connect."""
        return False

    mock_client.connect = _connect
    mock_client.is_connected = _is_connected

    coordinator = mock_entry.runtime_data.coordinator

    try:
        await coordinator._async_update_data()
    except UpdateFailed:
        await coordinator.async_shutdown()


async def test_setup_disconnect_five(
    hass: HomeAssistant,
    device_registry: dr.DeviceRegistry,
    mock_client: Mock,
    mock_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test setup creates expected devices."""

    async def _connect_success(self, *args, **kwargs) -> bool:
        """Mock BleakClient.connect."""
        return True

    mock_client.connect = _connect_success

    mock_entry.add_to_hass(hass)
    await hass.config_entries.async_setup(mock_entry.entry_id)
    await hass.async_block_till_done()

    assert mock_entry.state is ConfigEntryState.LOADED

    def _is_connected(self, *args, **kwargs) -> bool:
        """Mock BleakClient.is_connected."""
        raise BleakError

    mock_client.is_connected = _is_connected

    coordinator = mock_entry.runtime_data.coordinator

    with contextlib.suppress(UpdateFailed):
        await coordinator._async_update_data()


async def test_setup_invalid_mower_activity(
    hass: HomeAssistant,
    device_registry: dr.DeviceRegistry,
    mock_client: Mock,
    mock_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test setup creates expected devices."""

    def _is_connected_success(self, *args, **kwargs) -> bool:
        """Mock BleakClient.is_connected."""
        return True

    async def _connect_success(self, *args, **kwargs) -> bool:
        """Mock BleakClient.connect."""
        return True

    mock_client.connect = _connect_success
    mock_client.is_connected = _is_connected_success

    mock_entry.add_to_hass(hass)
    await hass.config_entries.async_setup(mock_entry.entry_id)
    await hass.async_block_till_done()

    assert mock_entry.state is ConfigEntryState.LOADED

    async def _mower_activity(self, *args, **kwargs):
        return None

    og_mower_activity = mock_client.mower_activity
    mock_client.mower_activity = _mower_activity

    coordinator = mock_entry.runtime_data.coordinator

    with contextlib.suppress(UpdateFailed):
        await coordinator._async_update_data()

    mock_client.mower_activity = og_mower_activity


async def test_setup_invalid_mower_state(
    hass: HomeAssistant,
    device_registry: dr.DeviceRegistry,
    mock_client: Mock,
    mock_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test setup creates expected devices."""

    def _is_connected_success(self, *args, **kwargs) -> bool:
        """Mock BleakClient.is_connected."""
        return True

    async def _connect_success(self, *args, **kwargs) -> bool:
        """Mock BleakClient.connect."""
        return True

    mock_client.connect = _connect_success
    mock_client.is_connected = _is_connected_success

    mock_entry.add_to_hass(hass)
    await hass.config_entries.async_setup(mock_entry.entry_id)
    await hass.async_block_till_done()

    assert mock_entry.state is ConfigEntryState.LOADED

    async def _mower_state(self, *args, **kwargs):
        return None

    og_mower_state = mock_client.mower_state
    mock_client.mower_state = _mower_state

    coordinator = mock_entry.runtime_data.coordinator

    with contextlib.suppress(UpdateFailed):
        await coordinator._async_update_data()

    mock_client.mower_state = og_mower_state


async def test_setup_invalid_battery(
    hass: HomeAssistant,
    device_registry: dr.DeviceRegistry,
    mock_client: Mock,
    mock_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test setup creates expected devices."""

    def _is_connected_success(self, *args, **kwargs) -> bool:
        """Mock BleakClient.is_connected."""
        return True

    async def _connect_success(self, *args, **kwargs) -> bool:
        """Mock BleakClient.connect."""
        return True

    mock_client.connect = _connect_success
    mock_client.is_connected = _is_connected_success

    mock_entry.add_to_hass(hass)
    await hass.config_entries.async_setup(mock_entry.entry_id)
    await hass.async_block_till_done()

    assert mock_entry.state is ConfigEntryState.LOADED

    async def _battery_level(self, *args, **kwargs):
        return None

    og_battery_level = mock_client.battery_level
    mock_client.battery_level = _battery_level

    coordinator = mock_entry.runtime_data.coordinator

    with contextlib.suppress(UpdateFailed):
        await coordinator._async_update_data()

    mock_client.battery_level = og_battery_level


async def test_setup_exception_battery(
    hass: HomeAssistant,
    device_registry: dr.DeviceRegistry,
    mock_client: Mock,
    mock_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test setup creates expected devices."""

    def _is_connected_success(self, *args, **kwargs) -> bool:
        """Mock BleakClient.is_connected."""
        return True

    async def _connect_success(self, *args, **kwargs) -> bool:
        """Mock BleakClient.connect."""
        return True

    mock_client.connect = _connect_success
    mock_client.is_connected = _is_connected_success

    mock_entry.add_to_hass(hass)
    await hass.config_entries.async_setup(mock_entry.entry_id)
    await hass.async_block_till_done()

    assert mock_entry.state is ConfigEntryState.LOADED

    async def _battery_level(self, *args, **kwargs):
        raise BleakError

    og_battery_level = mock_client.battery_level
    mock_client.battery_level = _battery_level

    coordinator = mock_entry.runtime_data.coordinator

    with contextlib.suppress(UpdateFailed):
        await coordinator._async_update_data()

    mock_client.battery_level = og_battery_level
