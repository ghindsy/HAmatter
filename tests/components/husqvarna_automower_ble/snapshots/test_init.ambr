# serializer version: 1
# name: test_setup
  None
# ---
